### ISSUE TITLE
Micro Endpoint does not allow wildcard subjects

### ISSUE THREAD

**Issue by scohen-examity** [nats-io/nats.go#1280]
## Feature Request
Allow the use of wildcards in an Endpoint definitions

#### Use Case:
A single handler may handle multiple operations (for instance, CRUD) and it's a convenience to be able to register
subscription subject like with the core nats subscriptions that allow wildcards.  Also, for recording/reporting handlers, wildcards
would be useful, e.g,. this is not possible with the current implementation:

	evtsGroup := srv.AddGroup("EVENTS")
	evtsGroup.AddEndpoint("ai.>", micro.HandlerFunc(aiEventHandler))
        evtsGroup.AddEndpoint("video.>", micro.HandlerFunc(videoEventHandler))

Where the respective handlers may be capturing payloads and updating some backend store -- having to enumerate
every specific event and add an endpoint to the same handler would add quite a bit of noise into the code.

#### Proposed Change:
Allow wildcards in Endpoint definitions like the current Subscribe behaviors

#### Who Benefits From The Change(s)?
Developers wanting to use the new Micro feature but maintain succinct but clear code

#### Alternative Approaches
Add an "AddSubscription" interface that essentially uses the underlying Subscriptions under the hood and leave Endpoint for precise configurations

**Comment by piotrpio** [comment]
Hey @scohen-examity , thanks for creating the issue.

As @Ol1BoT noticed, you can use endpoints with wildcards by using `WithEndpointSubject("foo.*")` option when adding the endpoint. The first parameter to `AddEndpoint()` is endpoint name, which cannot contain wildcards and if `WithEndpointSubject()` is not provided as option, it is also by default used as subject.

I agree that this is not clear - we'll at least improve the documentation or maybe even clean up the API a little bit to make the behavior more obvious.

**Comment by scohen-examity** [comment]
Thanks, I'll try that -- thanks for the help

**Comment by Karimerto** [comment]
I noticed another issue, likely the same cause as this. Not only are wildcards not allowed, but any multi-part subjects are not allowed either. Simply trying for example `AddEndpoint("some.service", micro.HandlerFunc(myfunc))` does not work either. You have to do:

    grp := srv.AddGroup("some")
    grp.AddEndpoint("service", micro...)

It's not really a big deal, but certainly at least the documentation should be corrected.

**Comment by ripienaar** [comment]
See WithEndpointSubject()

**Comment by Karimerto** [comment]
Yes, I realized as much after testing, but the initial documentation under "Endpoints and groups" is still incorrect :smile:
